### ISSUE TITLE
JavaScript Aggregation

### ISSUE THREAD

**Issue by hlship** [hlship/cascade#14]
Cascade should have a mechanism for adding assets (context or classpath) to the Javascript profile of a page. Further, it should be possible to combine multiple JavaScript libraries into fewer (ideally one or two) virtual assets, combining the contents of the underlying files.

**Comment by hlship** [comment]
Eventually, it would be nice to (optionally?) integrate a JavaScript optimizer.

**Comment by hlship** [comment]
Another part of this is aggregating generated JavaScript into a single block (again, patterned after Tapestry).

**Comment by hlship** [comment]
Note: both of these imply that Cascade knows what JS (either included libraries, or generated initialization JS) is needed before rendering the final response. 

This implies that in between rendering the DOM, and streaming it as the response, there's a transformation stage that can modify the DOM (adding the necessary <script> tags).

Further, where parts of the DOM are lazily rendered, or even deferred (rendered in parallel), it may be necessary to fully "flush" the DOM into existence, to fully identify what JavaScript needs to be aggregated across the entire document.

**Comment by hlship** [comment]
Fixed in ef0a86ababc79b82edc2d939622076d8a7767523
